### PR TITLE
Add scaling test

### PR DESCRIPTION
**Pull request by babakks**
This PR adds the scaling test, thanks to our beloved friend, @mina1460, who did a great job. :heart: 

Fixes CSS-8121

### CODE REVIEW

File: tests/integration/test_charm.py [hunk 1]
```diff
@@ -36,4 +36,11 @@ async def test_charm_crash(self, ops_test: OpsTest):
...
         await self.test_livepatch_server(ops_test)
-        return
+
+    async def test_scale_up_and_down(self, ops_test: OpsTest):
+        """Scale the livepatch machine charm up and then down."""
+        await scale(application_name=APP_NAME, ops_test=ops_test, count=2)
+        await self.test_livepatch_server(ops_test, unit=0)
+        await self.test_livepatch_server(ops_test, unit=1)
+        await scale(application_name=APP_NAME, ops_test=ops_test, count=1)
+        await self.test_livepatch_server(ops_test, unit=0)
```
Comment (alesstimec) [inline tests/integration/test_charm.py:46]: should we also assert that we no longer have two units deployed?

**Review by alesstimec — APPROVED** [review]
LGTM with a question

File: tests/integration/test_charm.py (hunk 1 above)
Comment (kian99) [inline tests/integration/test_charm.py:46]: Any way we can assert that the server is responding to requests?

**Review by kian99 — APPROVED** [review]

File: tests/integration/test_charm.py (hunk 1 above)
Comment (babakks) [reply tests/integration/test_charm.py:46]: We can check for a `/debug/status` or `info` endpoint. Will add.

Comment (babakks) [reply tests/integration/test_charm.py:46]: Yeah, I'll add that.

Comment (babakks) [reply tests/integration/test_charm.py]: Actually, as I checked again, the `scale` function also takes care of that:

https://github.com/canonical/livepatch-machine-charm/blob/64197035b7b19dc2be1d470602092c5fc07a6c4e/tests/integration/helpers.py#L51

**Review by ale8k — APPROVED** [review]
nice!

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -20,20 +26,63 @@
...
-        """Perform GET request on the Livepatch unit."""
-        url = await get_unit_url(ops_test, application=APP_NAME, unit=0, port=8080)
-        logger.info("curling app address: %s", url)
-        response = requests.get(url, timeout=300, verify=False)  # nosec
+    async def assert_http(
+        self, ops_test: OpsTest, protocol: str, application: str, unit: int, port: int, path: str = ""
+    ):
+        """Call given HTTP endpoint and assert if the call was successful (HTTP 200)"""
+        u = await get_unit_url(ops_test, protocol=protocol, application=application, unit=unit, port=port)
+        u += ("/" + path) if path else ""
```
Comment (kian99) [inline tests/integration/test_charm.py:34]: Nit: I think we can use Python's [urlparse](https://stackoverflow.com/questions/8223939/how-to-join-absolute-and-relative-urls) package here.

**Review by kian99 — APPROVED** [review]

File: tests/integration/test_charm.py (hunk 2 above)
Comment (babakks) [reply tests/integration/test_charm.py:34]: I'll apply this in the follow-up PR (#35).

**Pull request merged**